src/NuGet.Clients/NuGet.CommandLine/Common/Console.cs

```diff
@@ -282,12 +297,55 @@ public void PrintJustified(int startIndex, string text, int maxWidth)
                     // Print it with the correct padding
                     Out.WriteLine((leftPadding > 0) ? content.PadLeft(leftPadding) : content);
 
+                    if (content.Length == text.Length)
+                    {
+                        break;
+                    }
+
                     // Get the next substring to be printed
-                    text = text.Substring(content.Length);
+                    text = text.Substring(newLineIndex == -1 ? content.Length : crlf ? content.Length + 2 : content.Length + 1);
                 }
             }
         }
 
+
+        private (int, bool) NewLineIndex(string text, int length)
```

Comment:
this feels fragile to me, can you please explain why Environment.NewLine doesn't work?
If there's a component that's writing a string with a custom new line then we should fix that instead cause that's already wrong.

Reply:
Actually... good point. Be interested to see if any tests fail going back to that env constant.

Reply:
Given that this is going to be used by a search command, which may output package descriptions from NuGet sources, then assuming that the newline will always be normalized means that the server must know what the correct line ending is for the calling client. Or the client has to round-trip the package description though TextReader, as I think that `ReadLine()` correctly handles all line endings on all platforms.

Perhaps it's easier if `PrintJustified` reads its input line by line using `StringReader`?

Reply:
Like Andy said, the server may return the description to the client using a different newline variable (say \n) than the client's environment uses (\r\n), so the Env.Newline won't recognize the newline characters in the string, and will treat it as a regular character when partitioning the string for the indented printing. @donnie-msft 4 out of the 5 tests I wrote did cause the previous implementation to fail because of this. For example, `abcd\nwxyz`, when printed with an indent of 2 in a window of width 9, returned with `abcd\nwx\r\n  yz\r\n`, instead of the correct `abcd\r\n  wxyz\r\n`.

@zivkan I'm not sure I understand why the ReadLine() method would be a better way to implement this. Could you elaborate on that please?

Reply:
using something like:

```cs
using (var stringReader = new StringReader(input))
{
  string line;
  while ((line = stringReader.ReadLine()) != null)
  {
    // use line here
  }
}
```

it's far fewer lines than this method, `NewLineIndex` is. Also, this method requires the caller to handle 3 cases:
* A new line was not found
* A new line was found, and was just a single character
* A new line was found, and was CRLF

The `ReadLine` version abstracts that detail away, so the caller (`PrintJustified`) doesn't need to know anything about it.

Also the logic of detecting single vs multi-character new lines becomes someone else's problem, so we have fewer potentials for bugs.

Reply:
I've made this change in the PrintJustified method. Thanks for the suggestion.